### PR TITLE
[Libbeat] Add more complete tests for opt parameters in ES output

### PR DESCRIPTION
**Pull request by ph**
This PR add additionnal test over the usager of the `parameters` options
in the Elasticsearch output:

- When preconfigured params are set without local params
- When preconfigured params are set with local params
- When no preconfigured params are configured but local are.
- When no preconfigured or local params are set.

The merge is also done close to the actual calls and will not be
executed if anything fails before the bulk request.

The test assertion is now more solid and take into consideration any
errors.

See discussion in #18318 and #18326

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: libbeat/esleg/eslegclient/bulkapi_mock_test.go [hunk 1]
```diff
@@ -186,11 +174,89 @@ func TestEnforceParameters(t *testing.T) {
...
+			require.Equal(t, errShort, err)
+			require.Equal(t, len(recParams), len(test.expected))
+
+			for k, v := range test.expected {
+				assert.Equal(t, recParams.Get(k), v)
+			}
+		})
+	}
+
+	// params := map[string]string{
```
Comment (ruflin) [inline libbeat/esleg/eslegclient/bulkapi_mock_test.go:256]: Leftover?

File: libbeat/esleg/eslegclient/bulkapi_mock_test.go [hunk 2]
```diff
@@ -186,11 +174,89 @@ func TestEnforceParameters(t *testing.T) {
...
+				},
+			}
+
+			_, _, err := client.Bulk(context.Background(), index, "type1", test.reqParams, body)
+			require.Equal(t, errShort, err)
+			require.Equal(t, len(recParams), len(test.expected))
+
+			for k, v := range test.expected {
+				assert.Equal(t, recParams.Get(k), v)
+			}
```
Comment (urso) [inline libbeat/esleg/eslegclient/bulkapi_mock_test.go:252]: `assert.Equal(t, test.expected, recParams)` should actually work.

Comment (ph) [reply libbeat/esleg/eslegclient/bulkapi_mock_test.go:252]: +/- the  `url.Values` is a `map[string][]string`, which I find a bit weird, where every accessor only deals with key/value.. :

Comment (urso) [reply libbeat/esleg/eslegclient/bulkapi_mock_test.go:252]: Does it make sense to add another check that `recParams` does not include additional fields not found in `test.exptected`?

Comment (ph) [reply libbeat/esleg/eslegclient/bulkapi_mock_test.go:252]: We already do that by comparing the `len` of the map, so the ` len` all items need to match.
https://github.com/elastic/beats/pull/18393/files#diff-7bf553dedc14bb868bc139267101d16cR248

**Pull request merged**